core/os/android/adb/commands.go

```diff
@@ -108,6 +108,7 @@ retry:
 				continue retry
 			default:
 				// Some output we weren't expecting.
+				time.Sleep(time.Millisecond * 100)
```

Comment:
Shouldn't this be outside of the inner loop?

Reply:
I was trying to work out why the retry label exists at all. The best guess I can make is that it was the intention of the original author to keep polling adb infinitely as long as it prints the restarting message. For all other messages we want to have a fixed timeout of N tries that are M seconds apart. I tried to keep that intention, though I thought about changing it as it seemed a little weird to me.

Reply:
My point is that this is inside the "for each line" loop. So if we get multiple lines of output, this makes it needlessly sleep after each unexpected line. I'm not saying we shouldn't sleep if we don't get any of the other cases, just that this sleep should be after we've processed all the lines. If the inner loop completes, we always took the default case, so just putting the sleep after the inner loop, will make us retry with a sleep in-between, which is what you mentioned above is fixing the underlying issue.

The retry label exists to quickly jump to the next try and ignore the remaining lines. In the current code, yes, you could just break out of the inner loop, but I guess it was written so that it would be easy to add something after the inner loop, which is what I'm proposing we do now.

To combine this with your comment above: should we treat no output with a successful return code as success, rather than try-again?